### ISSUE TITLE
Generated autoloaders class paths

### ISSUE THREAD

**Issue by dincho** [composer/composer#3642]
As mentioned in #1585, in some environments there is performance improvements if not using concatenation in ```autoload_classmap.php``` and ```autoload_namespaces.php``` generated files.

However this would break couple of use cases - using composer in phar packages and building an application before deploying to production server (with different directory structure).

The proposition is to add option/parameter to allow users to dump those files without concatenation. Users should be allowed to specify target "vendor" directory, so that builds for production can benefit from this feature too.

example:
```bash
composer install --optimize-autoloader --with-absolute-path=/path/to/production/vendor
```

**Comment by sebastien-fauvel** [comment]
As discussed in #1585, this is my workaround on OpenShift for now:
```bash
echo "Optimizing composer autoloader."
sed -i "s|\$vendorDir . '|'${OPENSHIFT_REPO_DIR}vendor|" ${OPENSHIFT_REPO_DIR}/vendor/composer/autoload_classmap.php
sed -i "s|\$baseDir . '/|'${OPENSHIFT_REPO_DIR}|" ${OPENSHIFT_REPO_DIR}/vendor/composer/autoload_classmap.php
```

**Comment by Seldaek** [comment]
Fixed by #5174
